veps/sig-storage/incremental-backup.md

```diff
@@ -0,0 +1,420 @@
+      ...
+    </disk>
+    <disk type='block' device='disk' model='virtio-non-transitional'>
+      <driver name='qemu' type='raw' cache='none' error_policy='stop' io='native' discard='unmap'/>
+      <source dev='/dev/datavolumedisk2' index='3'/>
+      ...
+    </disk>
+
+```
+
+To:
+```xml
+    <disk type='file' device='disk' model='virtio-non-transitional'>
+      <driver name='qemu' type='qcow2' cache='none' error_policy='stop' discard='unmap'/>
+      <source file='/run/kubevirt-private/libvirt/qemu/swtpm/datavolumedisk.qcow2' index='2'>
+        <dataStore type='file'>
+          <format type='raw'/>
+          <source file='/run/kubevirt-private/vmi-disks/datavolumedisk/disk.img' index='3'/>
+        </dataStore>
+      </source>
+      ...
+    </disk>
+    <disk type='file' device='disk' model='virtio-non-transitional'>
+      <driver name='qemu' type='qcow2' cache='none' error_policy='stop' discard='unmap'/>
+      <source file='/run/kubevirt-private/libvirt/qemu/swtpm/datavolumedisk2.qcow2' index='4'>
+        <dataStore type='block'>
+          <format type='raw'/>
+          <source dev='/dev/datavolumedisk2' index='5'/>
+        </dataStore>
+      </source>
+      ...
+    </disk>
+```
+
+### VirtualMachineBackupTracker CR
+
+`VirtualMachineBackupTracker` is a namespace-scoped Custom Resource designed to track the latest backup checkpoint per-VM, per-backup-solution basis.
+It is intended to be created by a backup provider or user for a specific VM to monitor and manage incremental backups associated with a given backup solution.
+`changedBlockTracking` FeatureGate must be enabled in order to create this CR.
+Checkpoints of this tracker will be prefixed with the backupTracker's name, followed by a timestamp representing when the backup was taken.
+
+**Spec:**
+- `source` <br>
+Specifies the VM that this backup tracker is associated with.
+
+**Status**
+- `latestCheckpoint` <br>
+The checkpoint of the latest backup, updated by the backup controller. This checkpoint will serve as the base for the next incremental backup when providing this tracker as a reference.
+This struct will have:
+- name
+- creationTime
+- xmlData
+
+* If no latestCheckpoint in the status, a full backup will be initiated and the latestCheckpoint will be updated with the full backup checkpoint.
+* If latestCheckpoint is in the status an incremental backup will be performed using the `latestCheckpoint` as the base for determining which changed blocks need to back up.
+
+*VirtualMachineBackupTracker
+```yaml
+apiVersion: backup.kubevirt.io/v1alpha1
+kind: VirtualMachineBackupTracker
+metadata:
+    name: my-backup-tracker
+    namespace: ns1
+spec:
+    source:
+        apiGroup: kubevirt.io
+        kind: VirtualMachine
+        name: my-vm
+status:
+    latestCheckpoint:
+      name: my-backup-tracker-2025-03-03T16:13:28Z
+      creationTime: "2025-03-03T16:13:28Z"
+      xmlData: |
```

Comment:
Why do we need the xmlData in the status?

Reply:
We need the checkpoint metadata to be able to redefine them, but I like your direction of just wrap libvirt checkpoint xml as a go struct and have it as the latestCheckpoint explicitly. Ill update the VEP accordingly.